### PR TITLE
Fixed a cleanup related bug in ShareBackupsAndClusterWithUser

### PR DESCRIPTION
**Pull request by mkoppal-px**
**What this PR does / why we need it**:
During the cleanup of `ShareBackupsAndClusterWithUser` there was a backup created by testuser which was getting deleted in the `JustAfterEach`, but even before the deletion could complete we are deleting the backup location as `px-central-admin`. Since `px-central-admin` user did not have access to the backup, backup location delete request was throwing error even though we have set the param `DeleteBackups: true`.
To mitigate this, I have added a helper function `DeleteBackupAndWait` which will wait till the backup created by the testuser is deleted and then proceed with backup location deletion.

- Created `DeleteBackupAndWait` helper function
- `DeleteBackupLocation` function used in `CleanupCloudSettingsAndClusters` was not handling the error, so fixed that as well

**Which issue(s) this PR fixes** (optional)
Closes #PA-719

**Special notes for your reviewer**:
[Jenkins run](https://jenkins.pwx.dev.purestorage.com/job/Users/job/Sumit/job/Custom-Pipelines/job/px-backup-on-demand-system-test-byoc/442/consoleFull)

### CODE REVIEW

File: tests/backup/backup_basic_test.go [hunk 1]
```diff
@@ -2,11 +2,6 @@ package tests
 
 import (
 	"fmt"
-	"os"
-	"strings"
```
Comment (kshithijiyer-px) [inline tests/backup/backup_basic_test.go:6]: Why are we moving it down?

Comment (mkoppal-px) [reply tests/backup/backup_basic_test.go:6]: This is Goland's way for formatting.

Comment (kshithijiyer-px) [reply tests/backup/backup_basic_test.go:6]: But ideally standard libs should be on top right?

Comment (mkoppal-px) [reply tests/backup/backup_basic_test.go:6]: I ran the go formatter again and it did not show any change in this file

**Review by snigdha-px — APPROVED** [review]
LGTM

File: tests/backup/backup_basic_test.go (hunk 1 above)
Comment (kshithijiyer-px) [reply tests/backup/backup_basic_test.go:6]: Okay but this is a concern, we would need to find out more here. I see that in the PRs it keeps moving it up and down.

Comment (mkoppal-px) [reply tests/backup/backup_basic_test.go:6]: Yes. I agree. Even I am seeing this frequently.

**Review by lsrinivas-pure — APPROVED** [review]
lgtm

**Pull request merged**